tests/unit/SegmentQueryTest.php

```
@@ -0,0 +1,78 @@
+    }
+
+    public function testInvalidTypedValueFailsBeforeReplay(): void
+    {
+        $segment = SegmentDefinition::fromArray(array('id' => 'test', 'name' => 'Test', 'filters' => array(array('column' => 'amount', 'operator' => 'is', 'value' => 'not a number'))));
+        $this->expectException(InvalidArgumentException::class);
+        SegmentRequest::compile($segment, array($this->column()), $this->adapters());
+    }
+
+    public function testDisabledOperatorFailsBeforeReplay(): void
+    {
+        $segment = SegmentDefinition::fromArray(array('id' => 'test', 'name' => 'Test', 'filters' => array(array('column' => 'amount', 'operator' => 'gt', 'value' => '0'))));
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\Noteware\AdminTables\Segment\SegmentDefinition' in method 'testDisabledOperatorFailsBeforeReplay'. (undefined)

(StaticAccess)